[kokseng1] ip

Kokseng1 opened the pull request:
## Chatty
> "Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
-  Text-based
* Easy to use
+ ~~fast~~ __SUPER FAST__ to use

All you need to do is
1. Download it from [here](https://github.com/Kokseng1/ip)
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you :+1:


And it is **FREE**!

Features :
- [x] Managing tasks
- [x] Editing tasks
- [ ] GUI (coming soon)


If you Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
       Duke.run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Kokseng1 committed:
lvl0

Kokseng1 committed:
lvl1

Kokseng1 committed:
lvl2

Kokseng1 committed:
lvl3

Kokseng1 committed:
lvl4

Kokseng1 committed:
A-TextUiTesting

Kokseng1 committed:
A-TextUiTesting

Kokseng1 committed:
lvl6

Kokseng1 committed:
add taskfile

Kokseng1 committed:
lvl 7

Kokseng1 committed:
change if else statement to switch in Duke

Kokseng1 committed:
lvl 8

Kokseng1 committed:
A-MoreOOP

Kokseng1 committed:
A-Packages

Kokseng1 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

Kokseng1 committed:
A-Gradle

Kokseng1 committed:
A-JUnit

Kokseng1 committed:
Level-9

Kokseng1 committed:
A-CheckStyle

jflam1301 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    /**
+     * keeps reading commands and responding accordingly, until command is "bye"
```
Perhaps start the sentence with an uppercase letter and end with a punctuation?

jflam1301 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,129 @@
+public class Parser {
+    /**
+     * if command is delete, use this method to get the index of task user wants to delete
+     * @param text original command
```
Maybe add an empty line between the description and parameter section to follow the Java coding standard?
It is the same for other JavaDoc comments too.

jflam1301 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,129 @@
+ */
+public class Parser {
+    /**
+     * if command is delete, use this method to get the index of task user wants to delete
```
Maybe start the first sentence with verbs to follow the Java coding standard?
It is the same for other JavaDoc comments too.

jflam1301 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,50 @@
+    /**
+     * scanner to read input
+     */
+    private static final Scanner sc = new Scanner(System.in);
```
Perhaps changing the characters to uppercase letters since it is final?

jflam1301 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,105 @@
+                    taskList.unmarkTask(tasknum);
+                    break;
+                }
+
+                case DEADLINE:
+                case TODO:
+                case EVENT:
+                    Task nextTask = Parser.parseTask(inputString, taskType);
+                    taskList.addTask(nextTask);
+                    break;
+
+
+                case DELETE: {
+                    taskList.deleteTask(Parser.getDeleteIndex(inputString));
+                    break;
+                }
+
```
Maybe be consistent with your spacings?

jflam1301 reviewed (commented):
LGTM!

Kokseng1 committed:
Level-10

syamfarh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,205 @@
+            TaskType taskType = Parser.parseType(input);
+            switch (taskType) {
+            case LIST: {
+                return TaskList.getListAsString();
```
I think you forgot to put break; for every case

syamfarh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,129 @@
+    public static Task parseTask(String text, Duke.TaskType type) throws DukeException {
+        switch (type) {
+        case TODO: {
+            return new Todo(false, text);
```
I think you missed the break; in the switch cases here

syamfarh reviewed (commented):
LGTM!

Kokseng1 committed:
A-Assertions

Kokseng1 committed:
A-CodeQuality

Kokseng1 committed:
Tasks are not related to each other, no categorisation being done

Hard to analyse tasks, especially if there are many of them

Let's create tags that lets the user categorise tasks

Kokseng1 committed:
UI is a bit too bland

UI needs to appealing since audience will use the chatbot frequently

Beautifying the UI using scenebuilder and CSS

Kokseng1 committed:
Update README.md

Kokseng1 committed:
Update README.md

Kokseng1 committed:
Merge pull request #2 from Kokseng1/Kokseng1-updateReadME

Update README.md
